AstSemantics.md

```diff
@@ -350,7 +352,7 @@ The same operations are available on 64-bit integers as the ones available for
 Floating point arithmetic follows the IEEE-754 standard, except that:
  - The sign bit and significand bit pattern of any NaN value returned from a
    floating point arithmetic operation other than `neg`, `abs`, and `copysign`
-   are computed nondeterministically. In particular, the "NaN propagation"
+   are not specified. In particular, the "NaN propagation"
```

Comment:
On Thu, Jun 18, 2015 at 3:59 PM, Dan Gohman <notifications@github.com>
wrote:

> In AstSemantics.md
> <https://github.com/WebAssembly/design/pull/200#discussion_r32787706>:
>
> > @@ -350,7 +352,7 @@ The same operations are available on 64-bit integers as the ones available for
> >  Floating point arithmetic follows the IEEE-754 standard, except that:
> >   - The sign bit and significand bit pattern of any NaN value returned from a
> >     floating point arithmetic operation other than `neg`, `abs`, and `copysign`
> > -   are computed nondeterministically. In particular, the "NaN propagation"
> > +   are not specified. In particular, the "NaN propagation"
>
> "Not specified" makes it unclear whether implementations are allowed to
> change their behavior over time. We changed this text to say
> "nondeterministic" since there seems to be a consensus that running
> programs could migrate from one CPU to another with a different
> implementation, and there doesn't seem to be any clean way to handle that
> in the semantics.
>
Well, we do say "any NaN value returned from a floating point arithmetic
operation", which with the change to "not specified" should imply that
different operations could return different bit patterns.

The motivation for this change is that "are computed nondeterministically"
could be read as prescriptive, implying that the engine _must_ compute the
bits nondeterministically.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebAssembly/design/pull/200/files#r32787706>.
>